### PR TITLE
bind and other updates

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -21,23 +21,24 @@ class RunTask {
         this.runOne(taskItem, data, eachDone);
       }, done);
     }
-    const fn = this.tasks[task];
+    let fn = this.tasks[task];
+    if (this.options.bind) {
+      if (typeof fn === 'function') {
+        fn = fn.bind({ blah: '123' });
```
Comment (jgallen23) [inline index.js]: might want to check this

File: index.js [hunk 2]
```diff
@@ -21,23 +21,24 @@ class RunTask {
...
       return done(new Error(`${task} does not exist`));
     }
-    if (typeof fn.execute === 'function') {
-      onStart(task);
-      return fn.execute(data, (err, result) => {
-        onFinish(task);
-        return done(err, result);
-      });
-    }
+    fn = fn.execute ? fn.execute : fn;
```
Comment (jgallen23) [inline index.js]: lets remove this

File: index.js [hunk 3]
```diff
@@ -21,23 +21,24 @@ class RunTask {
         this.runOne(taskItem, data, eachDone);
       }, done);
     }
-    const fn = this.tasks[task];
+    let fn = this.tasks[task];
+    if (this.options.bind) {
```
Comment (jgallen23) [inline index.js]: this logic should be moved down after the array check, then you don't need the function check

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: index.js (hunk 2 above)
Comment (orthagonal) [reply index.js]: remove the check for fn.execute?  don't we need to be able to run it whether it's a runkit class or just a function?

Comment (jgallen23) [reply index.js]: we wrap the call on the runkit side anyways, no need to tie the two together

File: index.js [hunk 4]
```diff
@@ -21,23 +21,21 @@ class RunTask {
...
     if (Array.isArray(fn)) {
       return this.runOne(fn, data, done);
     }
-    onStart(task);
-    fn(data, done);
-    onFinish(task);
+    if (this.options.bind) {
+      fn = fn.bind(this.options.bind);
+    }
+    if (!fn) {
```
Comment (jgallen23) [inline index.js:28]: this needs to be above bind

**Review by jgallen23 — CHANGES_REQUESTED** [review]

**Pull request merged**